### DISCUSSION TITLE
gradient optimizer for PyTorch Temporal Fusion Transformer (TFT)

### DISCUSSION THREAD

**Discussion by grosestq** [Lightning-AI/pytorch-lightning#17036] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Dear Community,

I am using the PyTorch TemporalFusionTransformer implementation along with the PyTorch lightning Trainer. (https://pytorch-forecasting.readthedocs.io/en/latest/api/pytorch_forecasting.models.temporal_fusion_transformer.TemporalFusionTransformer.html + https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html)

Unfortuantely, i can't find any information about the default learning rate optimizer is used. Does anyone have any idea?

As far as I understand it correctly, the authors of TFT use the Adam optimizer with learning_rate and max_gradient_norm:
https://github.com/google-research/google-research/blob/master/tft/libs/tft_model.py

Best regards

Stefan

**Comment by grosestq** [comment, marked as answer]
I finaly found it :)
Pytorch uses the Ranger optimizer as default.
